src/dao/user.dao.ts

```diff
@@ -28,6 +28,31 @@ class UserDAO {
     });
   }
 
+  async fetchUsers(
+    filter: { username?: string; email?: string },
+    page = 1,
+    pageSize = 10
+  ) {
+    const skip = (page - 1) * pageSize;
+    const hasFilters = Object.values(filter).some((value) => value);
+
+    const whereCondition = [{
+      email: { contains: filter.email },
+    }, {
+      username: { contains: filter.username },
+    },];
+
+    const where: Prisma.UserWhereInput = {
+      OR: hasFilters ? whereCondition : undefined,
+    };
+
+    return prisma.user.findMany({
+      where: where,
+      skip,
+      take: pageSize,
+    });
```

Comment:
I think we can improve this query by iterating over the `filter` keys, and use it to push filter objects to the conditional:

```suggestion
    const skip = (page - 1) * pageSize;

    const whereCondition: (typeof Prisma.UserWhereInput.OR) = [];
    for (const key of Object.keys(filter)) {
      if (fitler[key] == null) continue;
      whereCondition.push({
        [key]: { contains: filter[key] }
      })
    }
    
    let where: Prisma.UserWhereInput = {};
    if (whereCondition.length > 0) {
      where = { OR: whereCondition };
    }

    return prisma.user.findMany({
      where,
      skip,
      take: pageSize,
    });
```

 LMK WDYT

Reply:
Yeah, I was thinking on something about that. I will update the query now, probably implementing something like that, I will let you know